src/plugins/discover/public/application/components/sidebar/discover_field_details.test.tsx

```diff
@@ -49,16 +60,185 @@ const indexPattern = getStubIndexPattern(
+    mockGetTriggerCompatibleActions.mockReturnValue([
+      {
+        getHref: mockGetHref,
+      },
+    ]);
+  });
+
+  function mountComponent(field: IndexPatternField, props?: Record<string, any>) {
+    const compProps = { ...defaultProps, ...props, field };
     return mountWithIntl(<DiscoverFieldDetails {...compProps} />);
   }
 
-  it('should enable the visualize link for a number field', function () {
+  it('should render buckets if they exist', async function () {
+    const visualizableField = new IndexPatternField(
+      {
+        name: 'bytes',
+        type: 'number',
+        esTypes: ['long'],
+        count: 10,
+        scripted: false,
+        searchable: true,
+        aggregatable: true,
+        readFromDocValues: true,
+      },
+      'bytes'
+    );
+    const buckets = [1, 2, 3].map((n) => ({
+      display: `display-${n}`,
+      value: `value-${n}`,
+      percent: 25,
+      count: 100,
+    }));
+    const comp = mountComponent(visualizableField, {
+      details: { ...defaultProps.details, buckets },
+    });
+    expect(findTestSubject(comp, 'fieldVisualizeError').length).toBe(0);
+    expect(findTestSubject(comp, 'fieldVisualizeBucketContainer').length).toBe(1);
+    expect(findTestSubject(comp, 'fieldVisualizeBucketContainer').children().length).toBe(
+      buckets.length
+    );
+    // Visualize link should not be rendered until async hook update
+    expect(findTestSubject(comp, 'fieldVisualizeLink').length).toBe(0);
+    expect(findTestSubject(comp, 'fieldVisualize-bytes').length).toBe(0);
+
+    // Complete async hook
+    await act(async () => {
+      await nextTick();
+      comp.update();
+    });
+    expect(findTestSubject(comp, 'fieldVisualizeError').length).toBe(0);
+    expect(findTestSubject(comp, 'fieldVisualizeBucketContainer').length).toBe(1);
+    expect(findTestSubject(comp, 'fieldVisualizeBucketContainer').children().length).toBe(
+      buckets.length
+    );
+    expect(findTestSubject(comp, 'fieldVisualizeLink').length).toBe(1);
+    expect(findTestSubject(comp, 'fieldVisualize-bytes').length).toBe(1);
+  });
+
+  it('should only render buckets if they exist', async function () {
+    const visualizableField = new IndexPatternField(
+      {
+        name: 'bytes',
+        type: 'number',
+        esTypes: ['long'],
+        count: 10,
+        scripted: false,
+        searchable: true,
+        aggregatable: true,
+        readFromDocValues: true,
+      },
+      'bytes'
+    );
+    const comp = mountComponent(visualizableField);
+    expect(findTestSubject(comp, 'fieldVisualizeContainer').length).toBe(1);
+    expect(findTestSubject(comp, 'fieldVisualizeError').length).toBe(0);
+    expect(findTestSubject(comp, 'fieldVisualizeBucketContainer').length).toBe(0);
+    expect(findTestSubject(comp, 'fieldVisualizeLink').length).toBe(0);
+    expect(findTestSubject(comp, 'fieldVisualize-bytes').length).toBe(0);
+
+    await act(async () => {
```

Comment:
Just curious what does this act() function do?

Reply:
https://reactjs.org/docs/test-utils.html#act

In this case I introduced it to help distinguish between the component state before and after an async hook runs (see L109-113). There are probably other/better ways to test a case like this, but it was an existing pattern in the codebase. Probably something we want to document better for other similar tests.